syncapi/sync/requestpool.go

```diff
@@ -134,13 +146,51 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *authtype
 }
 
 func (rp *RequestPool) currentSyncForUser(req syncRequest, latestPos types.StreamingToken) (res *types.Response, err error) {
+	res = types.NewResponse()
+
+	since := types.NewStreamToken(0, 0)
+	if req.since != nil {
+		since = *req.since
+	}
+
+	// See if we have any new tasks to do for the send-to-device messaging.
+	events, updates, deletions, err := rp.db.SendToDeviceUpdatesForSync(req.ctx, req.device.UserID, req.device.ID, since)
+	if err != nil {
+		return nil, err
+	}
+
+	// Before we return the sync response, make sure that we take action on
+	// any send-to-device database updates or deletions that we need to do.
+	// Then add the updates into the sync response.
+	defer func() {
```

Comment:
Why are we doing this in a `defer`?

Reply:
Mostly from when I was shuffling things around - it is a defer no longer.